include/Dialect/Poly/IR/PolyTypes.td

```diff
@@ -24,4 +24,9 @@ def Polynomial : Poly_Type<"Polynomial", "poly"> {
   let assemblyFormat = "`<` $ring `>`";
 }
 
+def PolynomialLike: TypeOrContainer<Polynomial, "polynomial-like">;
```

Comment:
`ElementwiseMappable` doesn't change the signature but simply impose constraints on the Op:
```
/// An `ElementwiseMappable` op must satisfy the following properties:
///
/// 1. If any result is a vector (resp. tensor), then at least one operand must
/// be a vector (resp. tensor).
/// 2. If any operand is a vector (resp. tensor), then there must be at least
/// one result, and all results must be vectors (resp. tensors).
/// 3. The static types of all vector (resp. tensor) operands and results must
/// have the same shape.
/// 4. In the case of tensor operands, the dynamic shapes of all tensor operands
/// must be the same, otherwise the op has undefined behavior.
/// 5. ("systematic scalarization" property) If an op has vector/tensor
/// operands/results, then the same op, with the operand/result types changed to
/// their corresponding element type, shall be a verifier-valid op.
/// 6. The semantics of the op on vectors (resp. tensors) shall be the same as
/// applying the scalarized version of the op for each corresponding element of
/// the vector (resp. tensor) operands in parallel.
/// 7. ("systematic vectorization/tensorization" property) If an op has
/// scalar operands/results, the op shall remain verifier-valid if all scalar
/// operands are replaced with vectors/tensors of the same shape and
/// corresponding element types.
```

We could remove the trait from the ops, but tests will still pass.

The value of adding it is to hint generalized "tools" about the behavior of the Op, e.g. `Linalg/Transforms/Vectorization`.

Reply:
> but I've been fighting weird bazel issues for an hour now and decided to give up for the day...

@AlexanderViand-Intel need any help from our side? Feel free to chat me or file an issue